### PR TITLE
Kafkachannel crd consolidation

### LINKED ISSUE #104: Convert To eventing-contrib KafkaChannel CRD
Historically (currently) this project has had it's own KafkaChannel CRD.  As we move closer towards hopefully consolidating this implementation with the enventing-contrib/kafka implementation it would be good to consolidate and build our implementation around their KafkaChannel CRD.  This will make the eventual consolidation easier as well as raise any important distinctions between the two.

### PR DESCRIPTION
**Pull request by travis-minke-sap**
We've moved to the same KafkaChannel CRD as defined in knative eventing-contrib/kafka.  This version is NOT backwards compatible with prior releases and the knative-kafka KafkaChannel CRD. 
 To migrate you will need to tear-down any prior KafkaChannels and uninstall the CRD / Helm Chart and re-install the new Helm Chart with the new CRD.

The two CRDs were actually quite close with the following changes being required...
  - Golang int types changed for a couple spec fields.
  - The RetentionMillis spec field was removed and for the time being can only be configured system wide via the environment variables on the controller deployment.

The eventing-contrib KafkaChannel Status, though, was significantly different as it aligns with their architecture components which are inherently different from ours.  We have made the necessary changes to align with their status tracking as much as possible and to simply hardcode passing status in a couple other cases.  As the two implementations are brought together we will have to make changes on either/both sides of this as makes sense at the time.

This also means we're no longer generating the associated apis/ or client/ implementations which have been removed.  We are not coding against the corresponding classes as generated in the eventing-contrib repo.

We've also moved to the knative eventing and eventing-contrib release-0.14 branches for this work. 
This is a good step towards releasing our own v0.14.0 version, and resulted in the removal of a bunch of deprecated utilities.  Our implementation is, though, still based on CloudEvents V1 despite knative having moved onto V2 in the release-0.14 branch.  Subsequent work is planned to move our usage of CloudEvents to V2 prior to release v0.14.0.

There is some concern as to whether the prior implementation for forwarding the Traceparent header through the stream is still functional in this delivery.  We're deferring that work until the CloudEvent sdk-go uplift has been done as the underlying implementations have all changed.

Also should note that, in preparation for this work, PRs #106 and #017 were delivered to clean up the common component a bit.

### CODE REVIEW

File: components/channel/internal/test/channel.go [hunk 1]
```diff
@@ -19,24 +18,23 @@ func CreateChannelReference(name string, namespace string) eventingChannel.Chann
...
 
 // Utility Function For Creating A Test KafkaChannel (Knative-Kafka)
-func CreateKafkaChannel(name string, namespace string, ready corev1.ConditionStatus) *knativekafkav1alpha1.KafkaChannel {
-	return &knativekafkav1alpha1.KafkaChannel{
+func CreateKafkaChannel(name string, namespace string, ready corev1.ConditionStatus) *kafkav1alpha1.KafkaChannel {
+	return &kafkav1alpha1.KafkaChannel{
 		TypeMeta: v1.TypeMeta{
-			Kind:       constants.KafkaChannelKind,
-			APIVersion: knativekafkav1alpha1.SchemeGroupVersion.String(),
+			Kind:       "KafkaChannel",
```
Comment (eric-sap) [inline components/channel/internal/test/channel.go:24]: Why did we change from using a constant to a static string?  It's "just a test file" so not a big deal I suppose.

File: components/channel/Gopkg.toml [hunk 2]
```diff
@@ -38,14 +38,24 @@
...
+
+# Align With Knative Eventing-Contrib
+[[constraint]]
+  name = "knative.dev/eventing-contrib"
+  branch = "release-0.14"
+
+# Align With Knative Eventing-Contrib
+[[constraint]]
+  name = "github.com/cloudevents/sdk-go"
+  version = "v2.0.0-RC1"
```
Comment (phamilton) [inline components/channel/Gopkg.toml:59]: Lets not forget to bump it up to v2 when released.

**Review by phamilton — APPROVED** [review]
LGTM

File: components/channel/Gopkg.toml (hunk 2 above)
Comment (travis-minke-sap) [reply components/channel/Gopkg.toml:59]: 👍

File: components/channel/internal/test/channel.go (hunk 1 above)
Comment (travis-minke-sap) [reply components/channel/internal/test/channel.go:24]: It was the one spot where we we're depending on the common package and it's just a test and the constant in question wasn't used elsewhere, etc.  so .... yes a bit of a dumb-down but for good reason (severed ties between components, etc..)

**Pull request merged**